### PR TITLE
ENH: support addition with pyarrow string dtypes

### PR DESCRIPTION
**Pull request by jbrockmendel**
@mroeschke it wouldn't surprise me if you say you prefer to explicitly _not_ support addition this way, but this seems like a pretty big use case, and the _libs.ops functions are a pretty great fit.

### CODE REVIEW

File: pandas/core/arrays/arrow/array.py [hunk 1]
```diff
@@ -454,6 +460,17 @@ def _cmp_method(self, other, op):
         return BooleanArray(values, mask)
 
     def _evaluate_op_method(self, other, op, arrow_funcs):
+        pa_type = self._data.type
+        if (pa.types.is_string(pa_type) or pa.types.is_binary(pa_type)) and op in [
+            operator.add,
+            roperator.radd,
+        ]:
+            if is_scalar(other):
```
Comment (mroeschke) [inline pandas/core/arrays/arrow/array.py:468]: Would be good to use `pc.binary_join_elementwise` for this

```
length = self._data.length()
seps = [""] * length
if is_scalar(other):
    other = [other] * length
result = pc.binary_join_element_wise(self._data, other, seps)
return type(self)(result)
```

**Review by mroeschke — COMMENTED** [review]
Just a merge conflict, otherwise LGTM

**Comment by mroeschke** [comment]
```
=================================== FAILURES ===================================
________________________________ test_str_join _________________________________
[gw1] linux -- Python 3.8.16 /home/runner/micromamba/envs/test/bin/python3.8

    @pytest.mark.xfail(
        reason="TODO: StringMethods._validate should support Arrow list types",
        raises=AttributeError,
    )
    def test_str_join():
        ser = pd.Series(ArrowExtensionArray(pa.array([list("abc"), list("123"), None])))
>       result = ser.str.join("=")
```
raises should change to a `NotImplementedError` it appears (which is better IMO)

**Comment by mroeschke** [comment]
Nevermind it looks like this was fixed in https://github.com/pandas-dev/pandas/pull/51441

**Comment by mroeschke** [comment]
```
pandas/core/arrays/arrow/array.py:472: error: List item 0 has incompatible type "bytes"; expected "str"  [list-item]
```

**Comment by jbrockmendel** [comment]
greenish

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**